server/item/armour_trim.go

```diff
@@ -0,0 +1,48 @@
+package item
+
+import "github.com/df-mc/dragonfly/server/world"
+
+type ArmourTrim struct {
+	Template ArmourTrimTemplate
+	Material TrimMaterial
+}
+
+type TrimMaterial interface {
+	// TrimMaterial returns the material name used for reading and writing trim data.
+	TrimMaterial() string
+	// MaterialColor returns the color code used for internal text formatting. Use text.Colourf for proper formatting.
+	MaterialColor() string
+}
+
+// MaterialFromString returns a TrimMaterial from a string.
+func MaterialFromString(name string) TrimMaterial {
```

Comment:
I'd rename this function to start with the same name as the type it returns, so TrimMaterial from string or a different name if the interface is renamed